uaclient/cli/__init__.py

```diff
@@ -897,6 +897,10 @@ def action_api(args, *, cfg, **kwargs):
     if args.options and args.data:
         raise exceptions.CLIAPIOptionsXORData()
 
+    if args.data and args.data == "-":
+        if not sys.stdin.isatty():
```

Comment:
I'm curious why wee need the `isatty` check - what does that do?

Reply:
This is just to guarantee that we will only execute that path if we are being called through a "redirect".
But given your other comment about the integration test, let me see if this logic prevents `stdin` somehow

Reply:
I have updated the test and that logic doesn't block the approach

Reply:
I'm still not sure it's necessary - I think it would prevent something like typing:
```
$ sudo pro api.u.pro.attach.token.full_token_attach.v1 --data -
{
"token": token
}
<Ctrl-D>
```
interactively in a shell. which I think would work if we read stdin without this check.

But also that use-case doesn't seem all that important. I'm fine either way

Reply:
I would say if someone is typing it to the CLI and <Ctrl-D>ing , they need to type their token there.
If they do, they don't care about not having the token in the CLI at all, but only not having it in history?
I don't know, seems to me that if there is a real use case for this scenario we can remove the check later. But IMHO, this feature makes way more sense for non interactive inputs.